typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -141,12 +172,38 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
         await warpRoute[method]();
         if (tokenType === TokenType.collateral) {
           const wrappedToken = await warpRoute.wrappedToken();
-          const xerc20 = IXERC20__factory.connect(wrappedToken, this.provider);
           try {
+            const xerc20 = IXERC20__factory.connect(
+              wrappedToken,
+              this.provider,
+            );
             await xerc20['mintingCurrentLimitOf(address)'](warpRouteAddress);
             return TokenType.XERC20;
-            // eslint-disable-next-line no-empty
-          } catch {}
+          } catch (error) {
+            this.logger.debug(
+              `Warp route token at address "${warpRouteAddress}" on chain "${this.chain}" is not a ${TokenType.XERC20}`,
+              error,
+            );
+          }
+
+          try {
+            const fiatToken = IFiatToken__factory.connect(
+              wrappedToken,
+              this.provider,
+            );
+
+            // Simulate minting tokens from the warp route contract
+            await fiatToken.callStatic.mint(NON_ZERO_SENDER_ADDRESS, 1, {
```

Comment:
isn't mint() on every erc20 token?

should we use fiatToken specifc fns like minterAllowance()

https://github.com/circlefin/stablecoin-evm/blob/master/contracts/v1/FiatTokenV1.sol

Reply:
it seems circle specific
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/token/interfaces/IFiatToken.sol#L4 

perhaps the author only included this terse interface so we don't need to bring in FiatToken package completely

Reply:
discussed offline, FiatToken may be a misnomer and should probably be MintableToken